scripts/process/portfolio.py

```diff
@@ -1,23 +1,42 @@
+"""
+Script to implement asset pricing using dominance as indicator
+"""
+
 import pandas as pd
-from environ.constants import PROCESSED_DATA_PATH, DEPENDENT_VARIABLES
-from environ.utils.variable_constructer import ma_variable_columns
+from environ.process.asset_pricing.double_sorting import (
+    stable_nonstable_split,
+    asset_pricing,
+)
+from environ.constants import (
+    PROCESSED_DATA_PATH,
+    FIGURE_PATH,
+    DEPENDENT_VARIABLES,
+)
 
 
 # load the regression panel dataset
 reg_panel = pd.read_pickle(
     PROCESSED_DATA_PATH / "panel_main.pickle.zip", compression="zip"
 )
 
-reg_panel = ma_variable_columns(
-    reg_panel,
-    DEPENDENT_VARIABLES,
-    time_variable="Date",
-    entity_variable="Token",
-    rolling_window_ma=30,
-)
+# split the dataframe into stablecoin and non stablecoin
+df_panel_stablecoin, df_panel_nonstablecoin = stable_nonstable_split(reg_panel)
 
-#  find medium value of all tokens' 30 day moving average for each day
+# stable non-stable info dict
+stable_nonstable_info = {
+    "stablecoin": df_panel_stablecoin,
+    "nonstablecoin": df_panel_nonstablecoin,
+}
 
-reg_panel[[f"{var}_median" for var in DEPENDENT_VARIABLES]] = reg_panel.groupby("Date")[
-    DEPENDENT_VARIABLES
-].transform("median")
+# iterate through the dominance
+for panel_info, df_panel in stable_nonstable_info.items():
+    for dominance in DEPENDENT_VARIABLES:
+        for frequency in [14, 30]:
+            asset_pricing(
+                df_panel,
+                str(FIGURE_PATH / f"{panel_info}_{dominance}_{frequency}.pdf"),
```

Comment:
conver to str shouldn't be necessary if you allow Path (which should be the input type in the first place)

Reply:
Fixed.